src/textual/widgets/_datatable.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import ClassVar, Generic, TypeVar, cast
+
+from rich.padding import Padding
+from rich.segment import Segment
+from rich.style import Style
+from rich.text import Text, TextType
+
+from .._lru_cache import LRUCache
+from .._segment_tools import line_crop
+from .._types import Lines
+from ..geometry import Size
+from ..reactive import Reactive
+from ..scroll_view import ScrollView
+from ..widget import Widget
+
+CellType = TypeVar("CellType")
+
+
+@dataclass
+class Column:
+    label: Text
+    width: int
+    visible: bool = False
+    index: int = 0
+
+
+@dataclass
+class Cell:
+    value: object
+
+
+class Header(Widget):
+    pass
+
+
+class DataTable(ScrollView, Generic[CellType]):
```

Comment:
This class makes my heart beat faster :heart_eyes: 

I have a few questions/comments I wanted to leave her, I hope you don't mind :+1: